File: app/src/main/java/com/example/hytool/total_directory/dagger/moudle/CommonFragmentModule.java
package com.example.hytool.total_directory.dagger.moudle;



import com.example.hytool.total_directory.base.BaseView;
import com.example.hytool.total_directory.mvp.present.Home_v2Present;

import dagger.Module;
import dagger.Provides;

/**
 * @author Administrator : HYY
 * TIME    2019/11/4
 * 备注 :
 */
@Module
public class CommonFragmentModule {
    BaseView mBaseView;

    public CommonFragmentModule(BaseView baseView) {
        mBaseView = baseView;
    }

    @Provides
    public Home_v2Present home_v2present() {
        return new Home_v2Present(mBaseView);
    }

}

File: app/src/main/java/com/hyy/hytool/dagger/noudle/CommonFragmentModule.kt
package com.hyy.hytool.dagger.noudle

import com.hyy.hytool.base.BaseView
import com.hyy.hytool.mvp.present.Home_v2Present
import dagger.Module
import dagger.Provides

/**
 * @Author : HYY
 * 描述:
 * 时间:      2022/05/2022/5/31
 * 软件包名:   com.hyy.hytool.dagger.noudle
 */
@Module
class CommonFragmentModule(var mBaseView: BaseView) {

    @Provides
    fun home_v2present(): Home_v2Present? {
        return Home_v2Present(mBaseView)
    }
}